docs/pages/concepts/smart-account-client.mdx

```diff
@@ -1,35 +1,134 @@
-  Contracts](/wallets/smart-contracts/choosing-a-smart-account) package exports
-  client definitions for hoisted instances of the Smart Contract Accounts.
+  Note: Changing the account type will deploy a different account. If you've
+  already deployed an account for a user and want to change the underlying
+  account type, you'll need to upgrade it. Learn how to upgrade
+  [here](/wallets/smart-contracts/other-accounts/modular-account/upgrading-to-modular-account).
 </Tip>
 
-When creating a Smart Account Client, you have two options:
+**Using 7702**
 
-1. You can pass an account instance directly into the client constructor. Doing so will use that account for all of the actions you perform with the client, and you won't have to
-   pass in the account each time you call a method. This is known as **"account hoisting"**.
-2. You can create a client without passing in an account. As a result, you'll have to pass in an account instance to each method you call on the client. This is really useful if you
-   want to manage multiple instances of an account in your app, and want to use one client for all of them.
+To use EIP-7702, pass `"7702"`" to the `mode` in `accountParams`. This will delegate the connected EOA to a smart account. Read more [here](/wallets/transactions/using-eip-7702).
 
-Let's see an example:
+**Advanced options**
+| Usage | Parameter | Type |
+| --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Smart contract account implementation | `type` | One of `"ModularAccountV2"`, `"LightAccount"`, `"MultiOwnerLightAccount"`, `"MultiOwnerModularAccount"` |
+| Sponsor gas for users. Get your policy ID [here](https://dashboard.alchemy.com/gas-manager) | `policyId` | `string` |
+| Simulate user operations before sending (i.e. use [`alchemyUserOperationSimulator`](/wallets/reference/account-kit/infra/functions/alchemyUserOperationSimulator)) | `useSimulation` | `boolean` |
+| Custom middleware to run before requesting sponsorship | `customMiddleware` | `ClientMiddlewareFn` |
+| Override fee estimation middleware (if you are using our RPCs, it's important to use the default [`alchemyFeeEstimator`](/wallets/reference/account-kit/infra/functions/alchemyFeeEstimator)) | `feeEstimator` | `ClientMiddlewareFn` |
+| Override gas estimation middleware (the default middleware calls the underlying bundler RPC to `eth_estimateUserOperationGas`) | `gasEstimator` | `ClientMiddlewareFn` |
+| Override middleware that signs user operation(s) | `signUserOperation` | `ClientMiddlewareFn` |
```

Comment:
I don't think any of these are valid options - they won't change the behavior of the resulting `SmartWalletClient`.

Reply:
And actually, these aren't valid options to pass to the JS client at all